src/ImpactEvaluator.sol

```diff
@@ -7,6 +7,7 @@ contract ImpactEvaluator is AccessControl {
     struct Round {
         uint end;
         string[] measurementsCids;
+        address payable[] addresses;
```

Comment:
I don't know about smart contracts to be able to judge the decision. Perhaps @AmeanAsad can help?

Reply:
Maybe we can change `getParticipantScore(uint roundIndex, address participant)` to a function that loops over addresses until it finds the index to extract the scores. Much less efficient but since its only reading data its free so doesn't cost us gas money. That way we can just remove the mapping altogether and just return to using arrays.